### PR TITLE
Improve segmentation (.seg.nrrd) file reader robustness

### PR DESCRIPTION
**Pull request by lassoan**
Make the segmentation storage node read .seg.nrrd files that are created outside of Slicer (e.g., using slicerio Python package).
They may use different order of fields (therefore the file detection heuristics had to be tuned), some fields may be missing, and there could be minor syntax differences (e.g., Slicer always terminates a list with separator character, but other software may just use separators between components).

### CODE REVIEW

File: Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx [hunk 1]
```diff
@@ -862,35 +862,42 @@ int vtkMRMLSegmentationStorageNode::ReadBinaryLabelmapRepresentation(vtkMRMLSegm
...
-          {
-            vtkWarningToMessageCollectionMacro(this->GetUserMessages(), "vtkMRMLSegmentationStorageNode::ReadBinaryLabelmapRepresentation",
-              "Segment extent is missing for segment " << segmentIndex);
-            for (int i = 0; i < 6; i++)
+            int currentSegmentExtent[6] = { 0, -1, 0, -1, 0, -1 };
+            std::string currentExtentString;
+            if (this->GetSegmentMetaDataFromDicitionary(currentExtentString, dictionary, segmentIndexForExtent, KEY_SEGMENT_EXTENT))
             {
-              currentSegmentExtent[i] = imageExtentInFile[i];
+              GetImageExtentFromString(currentSegmentExtent, currentExtentString);
```
Comment (Sunderlandkyl) [inline Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx:873]: If the extent string is invalid, then the extent of the current segment will be skipped.
Should we instead switch to the imageExtentInFile if we encounter an invalid extent string?

Comment (lassoan) [reply Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx:873]: By invalid do you mean empty (something like [0, -1, 0, -1, 0, -1])? That would mean that the segment is empty and so it should not expand the extent.

Comment (Sunderlandkyl) [reply Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx:873]: No, I meant that if the string itself was invalid, and could not be correctly parsed.

Comment (lassoan) [reply Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx:873]: Indeed all these parsing codes are still not fully robust for invalid input. The changes in this commit improve things, so I would integrate it as is, and implement further improvements later if needed.

**Pull request merged**